numpy/lib/function_base.py

```diff
@@ -2602,7 +2602,7 @@ def blackman(M):
     Plot the window and the frequency response:
 
     >>> from numpy.fft import fft, fftshift
-    >>> window = np.blackman(51)
+    >>> window = np.blackman(51) + 1e-10  # eps shift to avoid division by zero
```

Comment:
What happened without this? Which line causes the divide by 0?

Reply:
this generates a window with the first element 0, and that's fed into `log10` which generates a warning

Reply:
"division by zero" is a weird way to describe `log10(0)`. How about `logarithm of zero`?

Reply:
It's worse than that, the first element is actually negative

Reply:
Given that clipping occurs on line 2622, I think it would be better to use the `errstate` context manager to silence the warning on the line that performs the `log`

Reply:
Good suggestion, that is cleaner. updated